src/Homework 5/Program.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+
+namespace ATM_Managment
+{
+    public class ATM
+    {
+        public delegate void Action(decimal balance, string action);
+        public event Action BalanceHandler;
+
+        private decimal _balance = 0;
+
+        public void AddFunds(decimal _count)
+        {
+            _balance += _count;
+            BalanceHandler?.Invoke(_balance, "deposite");
+        }
+        public void WithdrawFunds(decimal _count)
+        {
+            if (_count > _balance)
+            {
+                Console.WriteLine("You do not have enough funds to withdraw");
+            }
+            else
+            {
+                _balance -= _count;
+                BalanceHandler?.Invoke(_balance, "withdrawal");
+            }
+        }
+        public void ShowBalance()
+        {
+            BalanceHandler?.Invoke(_balance, "\"Show balance\"");
+        }
+
+    }
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            ATM atm = new ATM();
+            bool exit = false;
+            while (!exit)
+            {
+                Console.ForegroundColor = ConsoleColor.Green;
+                Console.WriteLine("Hello! \nPlease select an operation:\nA - to add funds, \nW - to withdraw funds, \nB - to show a balance, \nE - to exit\n");
+                Console.ResetColor();
+                atm.BalanceHandler += ActionInfo;
```

Comment:
Subscribing to event in cycle can lead to unexpected behavior, because the handle will be run as many times as it was subscribed, by this reason you should unsubscribe at the end of the cycle, but you can subscribe just before it just once.

Reply:
Thank you